src/analyzer.rs

```diff
@@ -289,19 +289,61 @@ impl<'run, 'src> Analyzer<'run, 'src> {
     Ok(())
   }
 
-  fn resolve_alias(
-    recipes: &Table<'src, Rc<Recipe<'src>>>,
-    alias: Alias<'src, Name<'src>>,
+  fn resolve_alias<'a>(
+    modules: &'a Table<'src, Justfile<'src>>,
+    recipes: &'a Table<'src, Rc<Recipe<'src>>>,
+    alias: Alias<'src, Namepath<'src>>,
   ) -> CompileResult<'src, Alias<'src>> {
+    let Alias {
+      attributes,
+      name,
+      target,
+    } = alias;
+    let target_str = format!("{target}");
+    let mut path = target.into_inner();
+    let target = path
+      .pop()
+      .expect("Internal error: Alias target path can't be empty");
+
+    let alias = Alias {
+      attributes,
+      name,
+      target,
+    };
+
+    let mut maybe_recipes = Some(recipes);
+    if !path.is_empty() {
+      maybe_recipes =
+        Self::traverse_nonempty_module_name_path(&path, modules).map(|justfile| &justfile.recipes);
+    }
+
     // Make sure the target recipe exists
-    match recipes.get(alias.target.lexeme()) {
+    match maybe_recipes.and_then(|recipes| recipes.get(target.lexeme())) {
       Some(target) => Ok(alias.resolve(Rc::clone(target))),
       None => Err(alias.name.token.error(UnknownAliasTarget {
         alias: alias.name.lexeme(),
-        target: alias.target.lexeme(),
+        target: target_str,
       })),
     }
   }
+
+  fn traverse_nonempty_module_name_path<'modules>(
```

Comment:
I think this can be made simpler if it's recursive. Maybe with this signature:

```
fn foo(
  path: &[Name],
  modules: &Table<Justfile>,
  recipes: &Table<Rc<Recipe>>,
) -> Result<Option<Rc<Recipe>>, ()>;
```

If the path is empty, then resolution has failed, and you can return `Err(())`.

If path is length one, then look in `recipes` for the target recipe.

Otherwise, try to find the next module to look in using the first item in `path`. If not found, then return an error. If found, then make the recursive call with `path[1..]`, and the submodules recipes and modules.

Reply:
Hmm I didn't see added value in having an `Err(())` and a `Ok(None)` so I made it just an Option with `None` indicating that the recipe was not found